src/framework/filterControl/_filterItem.scss

```diff
@@ -45,3 +47,11 @@
         opacity: 1;
       }
     }
+
+.filterFormDropdown--editing {
+  top: -26px;
+}
+
+  .filterFormDropdown--editing .filterFormDropdown__form {
+    width: calc(100% - 2px);
```

Comment:
This `width` isn't enough. Try add one status and then edit... this could be a structural thing, I'll come to that later.